more/pixels/pixelMintController.js

```diff
@@ -1,95 +1,114 @@
 import { makePrivateName } from '../../util/PrivateName';
+import { insist } from '../../util/insist';
 
 import { getString } from './types/pixel';
 
 export function makeMintController(assay) {
   // Map from purse or payment to the rights it currently
   // holds. Rights can move via payments
 
-  // purse/payment to amount
-  let rights = makePrivateName();
-
   // pixel to purse/payment
-  const pixelToPursePayment = new Map();
+  const pixelToAsset = new Map();
 
-  function setLocation(amount, pursePayment) {
-    // purse/payment is the key of rights
+  function recordPixelsAsAsset(amount, asset) {
+    // purse or payment is the key of rights
     amount = assay.coerce(amount);
     const pixelList = assay.quantity(amount);
     for (const pixel of pixelList) {
-      pixelToPursePayment.set(getString(pixel), pursePayment);
+      pixelToAsset.set(getString(pixel), asset);
     }
   }
 
+  function makeAssetController(type) {
+    // asset to amount
+    let assets = makePrivateName();
+    return {
+      updateAmount(asset, newAmount) {
+        assets.set(asset, newAmount);
+        recordPixelsAsAsset(newAmount, asset);
+      },
+      recordNew(asset, initialAmount) {
+        assets.init(asset, initialAmount);
+        recordPixelsAsAsset(initialAmount, asset);
+      },
+      getAmount(asset) {
+        return assets.get(asset);
+      },
+      getType() {
+        return `${type}`;
+      },
+      has(asset) {
+        return assets.has(asset);
+      },
+      destroyAll() {
+        assets = makePrivateName(); // reset completely
+      },
+    };
+  }
+
+  const purseController = makeAssetController('purse');
+  const paymentController = makeAssetController('payment');
+
+  function isPurse(asset) {
+    return purseController.has(asset);
+  }
+
+  function isPayment(asset) {
+    return paymentController.has(asset);
+  }
+
+  function getController(asset) {
+    if (purseController.has(asset)) {
+      return purseController;
+    }
+    if (paymentController.has(asset)) {
+      return paymentController;
+    }
+    throw new Error(
+      `asset ${asset.getName()} was not recognized as a purse or a payment`,
+    );
+  }
+
+  // This amount (must be nonfungible) will be forcibly taken out of
+  // all purses and payments that it is currently in. Destroy is
+  // outside of an assetController because it could affect purses or
+  // payments
   function destroy(amount) {
     // amount must only contain one pixel
     const pixelList = assay.quantity(amount);
-    // assume length === 1 for now
+    insist(pixelList.length === 1)`amount must contain exactly one pixel`;
 
     const pixel = pixelList[0];
     const strPixel = getString(pixel);
-    if (!pixelToPursePayment.has(strPixel)) {
-      return;
-    }
-    const location = pixelToPursePayment.get(strPixel);
+    insist(pixelToAsset.has(strPixel))`\
+      pixel ${strPixel} could not be found to be destroyed`;
```

Comment:
My practice is to not have any leading whitespace on the line after backquote-backslash. We should discuss which makes code more or less readable. I see both sides. But we should try to consistently use one style until we change our mind to consistently use another style.

Also, the indentation here is a semantic change. These leading spaces are part of the literal text passed along to `insist`.

Reply:
Good point. I'll make that change